patches/minecraft/net/minecraft/client/particle/ParticleEngine.java.patch

```diff
@@ -1,14 +1,23 @@
 --- a/net/minecraft/client/particle/ParticleEngine.java
 +++ b/net/minecraft/client/particle/ParticleEngine.java
-@@ -26,7 +_,7 @@
- public class ParticleEngine {
-     private static final List<ParticleRenderType> RENDER_ORDER = List.of(ParticleRenderType.SINGLE_QUADS, ParticleRenderType.ITEM_PICKUP, ParticleRenderType.ELDER_GUARDIANS);
-     protected ClientLevel level;
--    private final Map<ParticleRenderType, ParticleGroup<?>> particles = Maps.newIdentityHashMap();
-+    private final Map<ParticleRenderType, ParticleGroup<?>> particles = Maps.newTreeMap(net.minecraftforge.client.ForgeHooksClient.makeParticleRenderTypeComparator(RENDER_ORDER));
```

Comment:
A note, if you're killing off makeParticleRenderTypeComparitor then delete the method in the hooks class.